UnitedIncome/components
# App 5072 mc gonagall ts

### Description
Added types to McGonagall component.

Highlights: 
* A new `McGonagallCardProps` type that can be used as the type for the the props argument of the `McGonagall` component's `card` prop (for rendering cards on steps).
* 

### Testing Instructions
* check `components` repo tests
* check  `client` repo tests (once integration is complete. See PR)

Updating the associated spec file to ts and converting it to use RTL was split out to another ticket.

- [JIRA Ticket](https://unitedincome.atlassian.net/browse/app-5072)
